lib/web/apiserver.go

```diff
@@ -569,22 +570,23 @@ func (h *Handler) getUserContext(w http.ResponseWriter, r *http.Request, p httpr
 	return userContext, nil
 }
 
-func localSettings(authClient auth.ClientI, cap types.AuthPreference) (webclient.AuthenticationSettings, error) {
+func localSettings(cap types.AuthPreference) (webclient.AuthenticationSettings, error) {
 	as := webclient.AuthenticationSettings{
 		Type:         constants.Local,
 		SecondFactor: cap.GetSecondFactor(),
 	}
 
-	// Add U2F settings, if available.
-	u2fs, err := cap.GetU2F()
-	if trace.IsNotFound(err) {
-		// No U2F settings.
-		return as, nil
+	// U2F settings.
+	if u2f, _ := cap.GetU2F(); u2f != nil {
```

Comment:
the error handling is unclear here. Does it mean it's safe to ignore `cap.GetU2F()` errors and continue? The same for `cap.GetWebauthn`.

Reply:
It is safe to ignore, yes. If U2F of Webauthn aren't configured then we just don't return their corresponding stanzas in webclient.AuthenticationSettings.

Reply:
hm...not configured is not the same as `error`. I would rather handle the errors rather then rely on implicit information about `GetU2F()` internal implementation. This makes it harder to reason about security in the context of this function. Why not return an error instead?